File: Backend/Hotel_Managment_System/HotelDataAceess/Entiteis/Room.cs
using HotelDataAceess.Enums;
using System;
using System.Collections.Generic;

namespace HotelDataAceess.Entiteis;

public  class Room
{
    public int RoomId { get; set; }

    public int RoomTypeId { get; set; }

    public string RoomNumber { get; set; } = null!;

    public byte RoomFloor { get; set; }

    public decimal RoomSize { get; set; }

    public AvailabilityStatus AvailabilityStatus { get; set; }

    public bool IsSmokingAllowed { get; set; }

    public bool IsPetFriendly { get; set; }
    public string RoomImageUrl { get; set; } = null!;
    public string? AdditionalNotes { get; set; }
    public virtual ICollection<Booking> Bookings { get; set; } = new List<Booking>();

    public virtual RoomType RoomType { get; set; } = null!;
}

File: Hotel_System/hotel-managment-system/src/services/models/Room.ts
import type { RoomType } from "./RoomType";

export interface RoomView {
  roomId: number;
  roomTypeDto: RoomType;
  roomNumber: string;
  roomFloor: number;
  availabilityStatus: string;
  isSmokingAllowed: boolean;
  isPetFriendly: boolean;
}

export interface Room {
  roomId: number;
  roomTypeId: number;
  roomNumber: string;
  roomFloor: number;
  availabilityStatus: string;
  roomImageUrl: File | string;
  isSmokingAllowed: boolean;
  isPetFriendly: boolean;
  additionalNotes: string | null;
}
